plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -305,26 +305,40 @@ protected void renameTable(ConnectorSession session, Connection connection, Stri
         if (!remoteSchemaName.equals(newRemoteSchemaName)) {
             throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming tables across schemas");
         }
-
-        execute(connection, format(
-                "sp_rename %s, %s",
-                singleQuote(catalogName, remoteSchemaName, remoteTableName),
-                singleQuote(newRemoteTableName)));
+        verifyTableName(connection.getMetaData(), newRemoteTableName);
+        String fullTableFromName = DOT_JOINER.join(
+                quoteForSqlServer(catalogName),
+                quoteForSqlServer(remoteSchemaName),
+                quoteForSqlServer(remoteTableName));
+
+        try (CallableStatement renameTable = connection.prepareCall("exec sp_rename ?, ?")) {
+            renameTable.setString(1, fullTableFromName);
+            renameTable.setString(2, newRemoteTableName);
+            renameTable.execute();
+        }
     }
 
     @Override
     protected void renameColumn(ConnectorSession session, Connection connection, RemoteTableName remoteTableName, String remoteColumnName, String newRemoteColumnName)
             throws SQLException
     {
-        execute(connection, format(
-                "sp_rename %s, %s, 'COLUMN'",
-                singleQuote(remoteTableName.getCatalogName().orElse(null), remoteTableName.getSchemaName().orElse(null), remoteTableName.getTableName(), "[" + escape(remoteColumnName) + "]"),
-                "[" + newRemoteColumnName + "]"));
+        String columnFrom = DOT_JOINER.join(
+                quoteForSqlServer(remoteTableName.getCatalogName().orElseThrow()),
+                quoteForSqlServer(remoteTableName.getSchemaName().orElseThrow()),
+                quoteForSqlServer(remoteTableName.getTableName()),
+                quoteForSqlServer(remoteColumnName));
+        verifyColumnName(connection.getMetaData(), newRemoteColumnName);
+
+        try (CallableStatement renameColumn = connection.prepareCall("exec sp_rename ?, ?, 'COLUMN'")) {
+            renameColumn.setString(1, columnFrom);
+            renameColumn.setString(2, newRemoteColumnName);
+            renameColumn.execute();
+        }
     }
 
-    private static String escape(String name)
+    private String quoteForSqlServer(String name)
     {
-        return name.replace("'", "''");
+        return "\"" + name.replace("\"", "\"\"") + "\"";
```

Comment:
That's SQL-standard quoting.
Is `this.quoteForSqlServer` equivalent to `this.quoted`?

i'm fine keeping them separate (the purpose and context of use is different), but let's have some javadoc explaining what it does

Reply:
But actually idea is the same so maybe let's use this.quoted?

Reply:
up to youy